lib/zmq-kernel.coffee

```diff
@@ -306,30 +329,27 @@ class ZMQKernel extends Kernel
 
 
     destroy: ->
```

Comment:
Unfortunately, the use of the `shutdown` message is undermined by frontends like the notebook. When the notebook restarts a kernel, what actually happens is that the notebook kills the kernel process and an relaunches a new kernel.

Judging by [this proposal](https://github.com/jupyter/enhancement-proposals/pull/14), this behaviour won't change in the future.